lib/elixir/lib/system.ex

```diff
@@ -369,9 +369,39 @@ defmodule System do
   """
   @spec get_env(binary) :: binary | nil
```

Comment:
@josevalim unrelated to this PR, but I noticed that `binary` in

```elixir
@spec get_env(binary) :: binary | nil
```

differs from `String.t` in

```
@spec get_env() :: %{optional(String.t) => String.t}
```

Under the hood, on master, get_binary/1 uses `List.to_string` which is also specced with `String.t`. Is is worth fixing, setting: `@spec get_env(binary) :: String.t | nil`?